aten/src/ATen/cuda/CUDASparseBlas.h

```diff
@@ -315,6 +315,33 @@ template <>
 void bsrsm2_solve<c10::complex<double>>(
     CUSPARSE_BSRSM2_SOLVE_ARGTYPES(c10::complex<double>));
 
+#define CUSOLVER_LINEAR_SOLVE_ARGTYPES(scalar_t)                                 \
+    cusolverSpHandle_t &handle, int n, int nnzA, const cusparseMatDescr_t descrA, \
+    const scalar_t *csrValA, const int *csrRowPtrA, const int *csrColIndA,       \
+    const scalar_t *b, float tol, int reorder, scalar_t *x, int *singularity
+
+// #define CUSOLVER_LINEAR_SOLVE_ARGTYPES(scalar_t) \
+//     cusolverSpContext *&handle, int &n, int &nnzA, const at::cuda::sparse::CuSparseMatDescriptor &descrA, \
+//     const scalar_t *&handle, const int *&csrRowPtrA, const int *&csrColIndA, scalar_t &b, float& tol, int &reorder, \
+//     scalar_t *&x, int *singularity
+
+template <typename scalar_t>
+inline void linear_solve(CUSOLVER_LINEAR_SOLVE_ARGTYPES(scalar_t)) {
```

Comment:
I think it's better to call this function `csrlsvluHost`. There are several linear solvers in the library. Here we just provide templated wrappers and it's better to keep the original name and arguments order.

Reply:
This makes sense, I temporarily named it `linear_solve` as I was just testing, but thanks for the suggestion Ivan. :)